doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,378 @@
+
+```js
+{
+  "profiles": [
+    {
+      "updates": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+      "fontSize": 16,
+      "fontWeight": "thin"
+    },
+    {
+      "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+      "name": "Cool Profile",
+      "commandline": "powershell.exe",
+      "antialiasingMode": "aliased",
+      "fontWeight": "bold",
+      "scrollbarState": "hidden"
+    }
+  ],
+  "schemes": [
+    {
+        "name": "Postmodern Tango Light",
+
+        "cursorColor": "#FFFFFF",
+        "selectionBackground": "#FFFFFF",
+
+        "background": '#61D6D6',
+        "foreground": '#E74856',
+
+        "black" : "#0C0C0C",
+        "blue" : "#0037DA",
+        "cyan" : "#3A96DD",
+        "green" : "#13A10E",
+        "purple" : "#881798",
+        "red" : "#C50F1F",
+        "white" : "#CCCCCC",
+        "yellow" : "#C19C00",
+        "brightBlack" : "#767676",
+        "brightBlue" : "#3B78FF",
+        "brightCyan" : "#61D6D6",
+        "brightGreen" : "#16C60C",
+        "brightPurple" : "#B4009E",
+        "brightRed" : "#E74856",
+        "brightWhite" : "#F2F2F2",
+        "brightYellow" : "#F9F1A5"
+    }
+  ]
+}
+```
+
+### Creation and location(s) of the json files
+
+#### For apps installed through Microsoft store (or similar)
+
+For apps that are installed through something like the Microsoft Store, they will need to declare themselves to
+be an app extension or write a separate app extension. In the app extension, they will need to create a public
+folder and insert their json files into it. The specifics of how they can do this are provided in the
+[Microsoft Docs](https://docs.microsoft.com/en-us/windows/uwp/launch-resume/how-to-create-an-extension) for how
+to create an app extension, and I will replicate the necessary section here. 
+
+In the appxmanifest file of the package:
+
+```
+<Package
+  ...
+  xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
+  IgnorableNamespaces="uap uap3 mp">
+  ...
+    <Applications>
+      <Application Id="App" ... >
+        ...
+        <Extensions>
+          ...
+          <uap3:Extension Category="windows.appExtension">
+            <uap3:AppExtension Name="Microsoft.com.Terminal"
+                               Id="<id>"
+                               PublicFolder="Public">
+            </uap3:AppExtension>
+          </uap3:Extension>
+        </Extensions>
+      </Application>
+    </Applications>
+    ...
+</Package>
+```
+
+Note that the name field **must** be `Microsoft.com.Terminal` for us to detect this extension. The `Id` field
+can be filled out as the app wishes. The `PublicFolder` field should have the name of the folder, relative to
+the package root, where the `json` files they wish to share with us are stored (this folder is typically named
+`Public` but can be renamed as long as it matches the relevant folder).
+
+During our profile generation, we will probe the OS for app extensions with the name `Microsoft.com.Terminal`

```

Comment:
We could also say that json files for settings should be put into a `Fragments` subdirectory within the `public` folder, so in the future if apps want to share more data with us they can just put it elsewhere in the `public` folder rather than declare a new app extension (correspondingly, we won't need to declare ourselves a new app extension host either). Which do you prefer?

Reply:
I'm thinking a `fragments` subdirectory is the best plan. Synergizes well with my request below too.